Modules/@babylonjs/react-native-basekit/scripts/install.js

```diff
@@ -0,0 +1,227 @@
+const https = require("https");
+const zlib = require("zlib");
+const tar = require("tar");
+const path = require("path");
+const fs = require("fs");
+
+function getCachePathCandidates() {
+  return [
+    process.env.npm_config_babylon_binary_cache,
+    process.env.npm_config_cache,
+  ].filter(function(_) {
+    return _;
+  });
+}
+
+function getCached(package, binaryFilename) {
+  return [
+    getCachePathCandidates(),
+    "BabylonReactNative",
+    package.version,
+    binaryFilename,
+  ].join("/");
+}
+
+function useCachedFile(filePath) {
+  return new Promise((resolve, reject) => {
+    if (fs.existsSync(filePath)) {
+      const gunzip = zlib.createGunzip();
+      const untar = tar.extract({ cwd: process.cwd() }); // Extract to the current working directory
+
+      const sourceStream = fs.createReadStream(filePath);
+      sourceStream
+        .pipe(gunzip) // Decompress the .tar.gz file
+        .pipe(untar) // Extract the tar contents to the cwd
+        .on("error", (err) => {
+          reject(err);
+        })
+        .on("end", () => {
+          resolve(true);
+        });
+    } else {
+      resolve(false);
+    }
+  });
+}
+
+function getArgument(name) {
+  const flags = process.argv.slice(2),
+    index = flags.lastIndexOf(name);
+
+  if (index === -1 || index + 1 >= flags.length) {
+    return null;
+  }
+
+  return flags[index + 1];
+}
+
+function getBinaryUrl(package, binaryFilename) {
+  var site =
+    getArgument("--brn-binary-site") ||
+    process.env.BRN_BINARY_SITE ||
+    process.env.npm_config_brn_binary_site ||
+    (package.nodeBabylonConfig && package.nodeBabylonConfig.binarySite) ||
+    "https://github.com/BabylonJS/BabylonReactNative/releases/download";
+
+  return [site, package.version, binaryFilename].join("/");
+}
+
+function downloadExtractAndCache(url, cachedFilePath, cb) {
+  const options = {
+    followRedirects: true, // Follow HTTP 3xx redirects
+  };
+
+  https
+    .get(url, options, (response) => {
+      if ([301, 302, 303, 307, 308].includes(response.statusCode)) {
+        // If the response is a redirect, recursively call downloadExtractAndCache with the new URL
+        downloadExtractAndCache(response.headers.location, cachedFilePath, cb);
+      } else if (response.statusCode === 200) {
+        const gunzip = zlib.createGunzip();
+        const untar = tar.extract({ cwd: process.cwd() }); // Extract to the current working directory
+
+        // create cache directory to store download file
+        const dir = path.dirname(cachedFilePath);
+        try {
+          if (!fs.existsSync(dir)) {
+            fs.mkdirSync(dir, { recursive: true });
+          }
+        } catch (err) {
+          reject(err);
```

Comment:
Are all these `reject(err)` calls supposed to be `cb(err)`? It doesn't seem like there is a `reject` in this context since it is not `Promise` based. I wonder if it would be better to write this in TypeScript and have a step to compile it to JavaScript?

Reply:
I think it's be better to use TS. For a future PR? I'm pretty sure this new package will needed adjustments and bug fixes.